lib/ansible/modules/cloud/amazon/aws_api_gateway.py

```diff
@@ -0,0 +1,246 @@
+description:
+     - Allows for the management of API Gatway APIs
+     - Normally you should give the api_id since there is no other
+       stable guaranteed unique identifier for the API.  If you do
+       not give api_id then a new API will be create each time
+       this is run.
+     - Beware that there are very hard limits on the rate that
+       you can call API Gateway's REST API.  You may need to patch
+       your boto.  See https://github.com/boto/boto3/issues/876
+       and discuss with your AWS rep.
+     - swagger_file and swagger_text are passed directly on to AWS
+       transparently whilst swagger_dict is an ansible dict which is
+       converted to JSON before the API definitions are uploaded.
+version_added: '2.3'
+requirements: [ boto3 ]
+options:
+  api_id:
+    description:
+      - The ID of the API you want to manage
+    required: false
+  state:
+    description:
+      - NOT IMPLEMENTED Create or delete API - currently we always create
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  swagger_file:
+    description:
+      - JSON or YAML file containing swagger definitions for API
+    required: false
+  swagger_text:
+    description:
+      - Swagger definitions for API in JSON or YAML as a string direct from playbook
+    required: false
+  swagger_dict:
+    description:
+      - Swagger definitions API ansible dictionary which will be converted to JSON and uploaded
+    required: false
+  stage:
+    description:
+      - stage API should be deployed to
+    required: false
+author:
+    - 'Michael De La Rue (@mikedlr)'
+extends_documentation_fragment:
+    - aws
+    - ec2
+notes:
+   - a future version of this module will probably use tags or another
+     ID so that an API can be create only once.
+   - as an early work around an intermediate version will probably do
+     the same using a tag embedded in the API name.
+
+'''
+
+EXAMPLES = '''
+# Update API resources for development
+tasks:
+- name: update API
+  aws_api_gateway:
+    api_id: 'abc123321cba'
+    state: present
+    swagger_file: my_api.yml
+
+# update definitions and deploy API to production
+tasks:
+- name: deploy API
+  aws_api_gateway:
+    api_id: 'abc123321cba'
+    state: present
+    swagger_file: my_api.yml
+    stage: production
+    deploy_desc: Make auth fix available.
+'''
+
+RETURN = '''
+output:
+  description: the data returned by put_restapi in boto3
+  returned: success
+  type: dict
+  sample:
+    'data':
+      {
+          "id": "abc123321cba",
+          "name": "MY REST API",
+          "createdDate": 1484233401
+      }
+'''
+
+from ansible.module_utils.basic import AnsibleModule, traceback
+from ansible.module_utils.ec2 import ec2_argument_spec, get_aws_connection_info, boto3_conn, camel_dict_to_snake_dict
+
+import os
+
+try:
+    import botocore
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+try:
+    import boto3 # noqua: F401
```

Comment:
You can avoid the need to import boto3 by just importing HAS_BOTO3 from module_utils.ec2